CDEFramework/Libraries/LegacyDictionarySupport/HttpHandlers/GlossaryLinkHrefHandler.cs

```diff
@@ -0,0 +1,208 @@
+﻿using System;
+using System.Web;
+using System.Net.Http;
+using System.Collections.Generic;
+using System.Configuration;
+using System.Linq;
+using System.Text;
+using System.Text.RegularExpressions;
+using System.Threading.Tasks;
+using Newtonsoft.Json;
+
+using LegacyDictionarySupport.Configuration;
+
+namespace LegacyDictionarySupport.HttpHandlers
+{
+
+    public class GlossaryLinkHrefHandler : IHttpHandler
+    {
+        // implement IsReusable method
+        public bool IsReusable
+        {
+            get { return false; }
+        }
+
+        public void ProcessRequest(HttpContext context)
```

Comment:
This method is doing too many things.  It implements logic to determine what values to pass to the API, it calls the API (via another method), it calculates a path for redirection, and it outputs the fallback page.  There really need to be at least three other methods with this one responsible only for coordinating the order the others are called in.